### PR TITLE
Refactor ContractDebugServiceTest dynamic calls, nested calls enum

### LINKED ISSUE #8944: Refactor `ContractDebugServiceTest`
### Problem

The `ContractDebugServiceTest` needs to be refactored to use web3j and the new integration test approach such as `ContractCallDynamicCallsTest` in this [PR](https://github.com/hashgraph/hedera-mirror-node/pull/8912).

### Solution

Refactor `ContractDebugServiceTest` to be independent of `ContractCallTestSetup` and the other tests.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by kselveliev**
**Description**:
This pr refactors `ContractDebugServiceTest`. Migrated tests from `ContractDebugServiceTest` that uses `DynamicCallsContractFunctions`. Test were migrated into `ContractCallDynamicCallsTests` as it also tests `DynamicCallsContractFunctions`. We just add the opcode assertions in the already existing tests there since they use the same setup. This also decreases the time for all tests to run. Moved `NestedEthCallContractFunctionsNegativeCases` to the nested calls test classes and just added opcode tracer assertions in the specified test. 

This PR modifies:
`ContractDebugServiceTest` - removed `ContractDebugServiceTest` function.
`ContractCallDynamicCallsTests` - added debug trace assertions in the tests testing dynamic calls functions - uses exactly the same function of the removed `evmDynamicCallsTokenFunctions`.
`ContractCallNestedCallsTest` - added debug trace assertions in the tests.
`ContractCallNestedCallsHistoricalTest` - added debug trace assertions in the tests.
Deleted ContractDebugServiceTests as all test cases are moved to the above mentioned classes.

This pr is related to https://github.com/hashgraph/hedera-mirror-node/issues/8944 but there will be a couple of prs more in future before we can close this task.

**Related issue(s)**:

Fixes #8944 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallDynamicCallsTest.java [hunk 1]
```diff
@@ -833,4 +911,86 @@ private void nftAllowancePersist(
                         .approvedForAll(true))
                 .persist();
     }
+
+    private void verifyOpcodeTracerCall(final String callData, final ContractFunctionProviderRecord functionProvider) {
```
Comment (bilyana-gospodinova) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallDynamicCallsTest.java:915]: These  verifyOpcode* methods are used in here and in the other PR (https://github.com/hashgraph/hedera-mirror-node/pull/9175). They will possibly be used in other classes too. Can't we move them in some parent class?

Comment (kselveliev) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallDynamicCallsTest.java:915]: I thought about this. 
But are there are a few cons to it.
This method internally uses params captor, contextCaptor, and contractDebugService.
So to put them in a parent class i would need to be passing those 3 as parameters to the method and the method will have 5 parameters.
Also if someone want to use this method it needs to have handled beforehand the initialization of captor, contextCaptor, and contractDebugService. I think it gets a bit confusing that way.
I’m open to suggestions how to implement this in a clean way.

Comment (kselveliev) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallDynamicCallsTest.java]: Extracted the common logic in a parent class. This addressed the comment.

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallNestedCallsTest.java [hunk 2]
```diff
@@ -547,4 +593,57 @@ private Entity accountPersistWithBalance(final long balance) {
...
+                        result -> assertThat(result.getHaltReason())
+                                .isPresent()
+                                .map(ExceptionalHaltReason::getDescription)
+                                .hasValue(function.expectedErrorMessage()));
+        assertThat(actual.opcodes().size()).isNotZero();
+        assertThat(toHumanReadableMessage(actual.opcodes().getLast().reason()))
+                .isEqualTo(function.expectedErrorMessage());
+    }
+
+    private void verifySuccessfulOpcodeTracerCall(final ContractDebugParameters params) {
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallNestedCallsTest.java]: I also think we can extract these common verify trace methods in `AbstractContractCallServiceTest`. The `ContractDebugService` can be initialized there and the captors can be passed as arguments. You can add in a javadoc above the methods that the passed captors should be initialized before invoking these methods.

Comment (kselveliev) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/ContractCallNestedCallsTest.java]: I already extracted them in the latest commit.

File: hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/AbstractContractCallServiceOpcodeTracerTest.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+import org.hyperledger.besu.evm.frame.ExceptionalHaltReason;
+import org.junit.jupiter.api.BeforeEach;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.web3j.tx.Contract;
+
+@RequiredArgsConstructor
+abstract class AbstractContractCallServiceOpcodeTracerTest extends AbstractContractCallServiceTest {
+
+    private final ContractDebugService contractDebugService;
```
Comment (IvanKavaldzhiev) [inline hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/AbstractContractCallServiceOpcodeTracerTest.java:44]: I believe you can inject this service as a `@Resource` and get rid of the constructors in all class inheritants.

Comment (kselveliev) [reply hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/AbstractContractCallServiceOpcodeTracerTest.java]: comment addressed

**Review by bilyana-gospodinova — DISMISSED** [review]
LGTM

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Pull request merged**